### ISSUE TITLE
Issue to connect to HTTPS

### ISSUE THREAD

**Issue by mokchenda** [splunk/splunk-sdk-php#37]
Hello,
I'm having an issue to connect to a splunk server configured with HTTPS.
For information, using curl and python scripts against the same server work correctly for me.
Attached the stactrace exception:
    [message:protected] => fopen(https://splunk-api.crz.xxxx.com:443/services/auth/login): failed to open stream: operation failed
    [string:Exception:private] => 
    [code:protected] => 2
    [file:protected] => /Applications/MAMP/htdocs/splunk-sdk-php-master/Splunk/Http.php
    [line:protected] => 118
    [trace:Exception:private] => Array

Please advise and let me know if you need more information.
Thanks,
Chenda

**Comment by joseaugustocardoso** [comment]
I have the same problem

**Comment by tdhellmann** [comment]
Hi @joseaugustocardoso - which version of PHP are you using, and what's the exact error message?

**Comment by joseaugustocardoso** [comment]
PHP7
 fopen(https://myserverip:443/services/auth/login): failed to open stream: operation failed

**Comment by tdhellmann** [comment]
The version of PHP you're using is probably not compatible with the way the PHP SDK was written. As far as I know we built the PHP SDK on PHP 5.2/5.3, and you'll see comments to the effect that we're not handling certificate validation for HTTPS in the code. I would expect the error you're encountering is related to this. Splunk by default uses a self-signed certificate, which could be the problem. If so, you might be able to resolve this by getting a non-self-signed certificate or adding in certificate validation.

**Comment by falon** [comment]
With PHP7 neither 'http' doesn't work:

`fopen(http://splunk.example.com:8089/services/auth/login): failed to open stream: HTTP request failed!`

Are you planning a release for PHP7?
Many thanks

**Comment by tdhellmann** [comment]
You should have to connect to 8089 using HTTPS by default, so unless you've edited Splunk's configurations that could be your issue. 

We're not planning a release for PHP7 at this time.

**Comment by adamziaja** [comment]
Same there, problem is in Splunk...

```fopen(https://localhost:8089/services/auth/login): failed to open stream: operation failed```

```
$ curl -k https://localhost:8089/services/auth/login
<?xml version="1.0" encoding="UTF-8"?>
<response>
  <messages>
    <msg type="WARN">Login failed</msg>
  </messages>
</response>
```

```
$ php --version
PHP 7.0.8-3ubuntu3
```

**Comment by falon** [comment]
In my experience, now I can confirm that with PHP7 if I use "http" scheme it works.
I only had to remember to set
```
[sslConfig]
enableSplunkdSSL = false
```
in server.conf  ;)

**Comment by adamziaja** [comment]
Problem is in self-signed certificate, I made fix:
https://github.com/adamziaja/splunk-sdk-php/blob/master/Splunk/Http.php#L108-L111
This is secure to use on localhost.

Other way is to use certificate in way like:
```
'ssl'=>array(
    'verify_peer' => true,
    'cafile' => '/path/to/ca-bundle.crt'
)
```

**Comment by code-guru** [comment]
Pull request https://github.com/splunk/splunk-sdk-php/pull/41 will fix the issue with SSL certificates.

**Comment by adamziaja** [comment]
@code-guru WTF? so you copied my pull to yours? also allow_self_signed is no needed :)
https://github.com/splunk/splunk-sdk-php/pull/40/commits/56441f1290e5312f4c3de482538d85e3c1b9f209
https://github.com/splunk/splunk-sdk-php/pull/41/commits/313450b361f600315bd3c63a5496af9719607241

**Comment by shakeelmohamed** [comment]
This is fixed in the develop branch vida #41.
